src/lib.rs

```diff
@@ -26,11 +26,13 @@ mod common;
 pub mod concept;
 mod connection;
 mod database;
+pub mod logic;
 pub mod transaction;
 
 pub use self::{
     common::{error, Credential, Error, Options, Result, SessionType, TransactionType},
     connection::Connection,
     database::{Database, DatabaseManager, Session},
+    logic::{LogicManager, Rule},
```

Comment:
We don't need to expose `LogicManager` and `Rule` at the top level. `typedb_client::logic::Rule` seems better scoped, and `LogicManager` is only really accessed via a transaction instance.